core/domain/user_jobs_one_off_test.py

```diff
@@ -1585,6 +1585,272 @@ def test_one_setting_model_without_gae_id(self):
             migrated_setting_model.last_updated)
 
 
+class FixUserSettingsCreatedOnOneOffJobTests(test_utils.GenericTestBase):
+
+    USER_ID_1 = 'user_id'
+    USER_ID_2 = 'user_id_2'
+    EMAIL_1 = 'test@email.com'
+    EMAIL_2 = 'test2@email.com'
+    SKILL_ID_1 = 'skill_id_1'
+    SKILL_ID_2 = 'skill_id_2'
+    DEGREE_OF_MASTERY = 0.5
+    EXPLORATION_IDS = ['exp_1', 'exp_2', 'exp_3']
+    COLLECTION_IDS = ['col_1', 'col_2', 'col_3']
+    EXP_ID_ONE = 'exp_id_one'
+    EXP_ID_TWO = 'exp_id_two'
+    EXP_ID_THREE = 'exp_id_three'
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            user_jobs_one_off.FixUserSettingsCreatedOnOneOffJob.create_new())
+        user_jobs_one_off.FixUserSettingsCreatedOnOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_mapreduce_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_mapreduce_tasks()
+        stringified_output = (
+            user_jobs_one_off.FixUserSettingsCreatedOnOneOffJob
+            .get_output(job_id))
+        eval_output = [ast.literal_eval(stringified_item) for
+                       stringified_item in stringified_output]
+        return eval_output
+
+    def test_update_created_on_using_user_settings_model_only(self):
+        user_settings_model = (
+            user_models.UserSettingsModel(
+                id=self.USER_ID_1,
+                email=self.EMAIL_1,
+            )
+        )
+        user_settings_model.update_timestamps()
+        original_created_on_timestamp = user_settings_model.created_on
+        user_settings_model.created_on += datetime.timedelta(hours=10)
+        user_settings_model.last_started_state_editor_tutorial = (
+            original_created_on_timestamp + datetime.timedelta(hours=1))
+        user_settings_model.last_created_an_exploration = (
+            original_created_on_timestamp + datetime.timedelta(hours=2))
+
+        final_created_on_timestamp = user_settings_model.last_updated
```

Comment:
Could you add comment above this explaining the we did not modify `last_updated` and because of that it is now the minimal value. It took me around a minute to figure out why exactly `last_updated`.

Reply:
Sure, I have added a comment saying that now. Added one or two comments elsewhere too where I felt similar.